tests/lookup/tests.py

```diff
@@ -956,3 +957,15 @@ def test_nested_outerref_lhs(self):
             ),
         )
         self.assertEqual(qs.get(has_author_alias_match=True), tag)
+
+    def test_exact_selected_field_rhs_subquery(self):
+        """Ensures GROUP BY in query persists when used as subquery."""
+        _ = Author.objects.create(name='one')
+        author_2 = Author.objects.create(name='two')
+        max_ids = Author.objects.values('alias').annotate(m=Max('id')).values('m').order_by()
+        # Include empty .order_by() to disregard ordering set in Author's Meta class.
+
+        self.assertEqual('SELECT MAX("lookup_author"."id") AS "m" FROM "lookup_author" GROUP BY "lookup_author"."alias"', str(max_ids.query))
+        authors = Author.objects.filter(id=max_ids[:1])
+
+        self.assertEqual(authors[0], author_2)
```

Comment:
You could use `.get()` to assert a single result is returned

```suggestion
        self.assertEqual(authors.get(), author_2)
```